veggieseasons/lib/screens/details.dart

```diff
@@ -6,68 +6,222 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/widgets.dart';
 import 'package:scoped_model/scoped_model.dart';
 import 'package:veggieseasons/data/app_state.dart';
-import 'package:veggieseasons/data/veggie.dart';
+import 'package:veggieseasons/data/preferences.dart';
 import 'package:veggieseasons/styles.dart';
 import 'package:veggieseasons/widgets/close_button.dart';
 
-/// A circular widget that represents a season of the year.
-///
-/// The season can be displayed as a valid harvest time or one during which a
-/// particular veggie cannot be harvested. Bright colors are used in the first
-/// case, and grays in the latter.
-class SeasonCircle extends StatelessWidget {
-  const SeasonCircle(this.season, this.isHarvestTime);
+class InfoView extends StatelessWidget {
+  final int id;
 
-  /// Season to be displayed by this widget.
-  final Season season;
+  const InfoView(this.id);
 
-  /// Whether or not [season] should be presented as a valid harvest season.
-  final bool isHarvestTime;
+  // Creates a [Text] widget to display a veggie's "percentage of your daily
+  // value of this vitamin" data adjusted for the user's preferred calorie
+  // target.
+  Widget _buildVitaminText(int standardPercentage, Future<int> targetCalories) {
+    return FutureBuilder(
+      future: targetCalories,
+      builder: (context, snapshot) {
+        final target = snapshot?.data ?? 2000;
+        final percent = standardPercentage * 2000 ~/ target;
 
-  String get _firstChars {
-    return '${season.toString().substring(7, 8).toUpperCase()}'
-        '${season.toString().substring(8, 9)}';
+        return Text(
+          '$percent% DV',
```

Comment:
Is DV a common word people know?

Reply:
I've seen it in multiple sources while looking up nutrition facts. I'll update the note below the values to provide a hint, though.